### PR TITLE
bug/NQ-1354: Correct string parameter passing and parameter scope resolution

### PR DESCRIPTION
**Pull request by Jonah-Foley**
YouTrack ticket: https://nu-quantum.youtrack.cloud/issue/NQ-1354/Correct-top-level-string-parameter-passing

### CODE REVIEW

**Comment by Jonah-Foley** [comment]
Closing based on IRL discussion. The string escaping can be performed within the `.core` file itself. For example: 
```
parameters:
      - MemInitFileCh0='"../data/memory/pulse_ch0.mem"'
```
or
```
parameters:
      - MemInitFileCh0="\"../data/memory/pulse_ch0.mem\""
```

File: src/edalize/tools/flist.py [hunk 1]
```diff
@@ -26,20 +27,51 @@ class Flist(Edatool):
...
     }
 
     def setup(self, edam):
         super().setup(edam)
 
         self.f = []
 
+        simulator = self.tool_options.get("simulator", None)
+
+        if simulator == "None":
```
Comment (Jonah-Foley) [inline src/edalize/tools/flist.py:55]: Surprisingly it is not actually `None`, but a string of value `"None"`

File: src/edalize/tools/flist.py [hunk 2]
```diff
@@ -283,6 +320,13 @@ def get_parser():
         action="store_true",
         help="use verbose logging",
     )
+    parser.add_argument(
+        "-s",
+        "--simulator",
+        choices=Flist._sim_prefixes.keys(),
+        help="Name of the simulator tool which consumes the flist output",
+        required=False,
```
Comment (Jonah-Foley) [inline src/edalize/tools/flist.py:328]: `required=False`, Agree?

File: src/edalize/tools/flist.py (hunk 1 above)
Comment (shareefj) [reply src/edalize/tools/flist.py:55]: I've had a look at this and can't work out what's going on.  I'll ask on the forum once we've committed this and can point at it.

File: src/edalize/tools/flist.py (hunk 2 above)
Comment (shareefj) [reply src/edalize/tools/flist.py:328]: By default it's not required so it's technically redundant.

**Review by shareefj — APPROVED** [review]
Think it's fine.

**Pull request merged**